### PR TITLE
Crit Dental Pills!

### PR DESCRIPTION
**Pull request by LucyGrind**
## About The Pull Request
What it says on the tin: Allows dental pills to be used in crit, simply gets rid of hands_free and adds a NONE flag
## Why It's Good For The Game
Dental implants are among the lowest used surgeries in the game, and I think have a lot of potential for funny shenanigans, such as using ghoul powder to fake your own death, or just blowing yourself up when in crit. Requires another person to do the surgery, so its harder to powergame with and also reviver implants alongside other things set the precedent that this sorta change is okay.
## Changelog
:cl: Hacks-The-Doors
balance: dental pills can now be used in crit.
balance: dental pills now give off a message when you start using them
balance: dental pills have a 2.5 second delay when in soft crit
/:cl:

### CODE REVIEW

**Comment by Ryll-Ryll** [comment]
Not a fan of adding additional ways for people to stand back up and continue a fight that they've lost, especially when there's no visible message to people around them that you just bit a pill

**Comment by xXPawnStarrXx** [comment]
> Not a fan of adding additional ways for people to stand back up and continue a fight that they've lost, especially when there's no visible message to people around them that you just bit a pill

I agree with this, make it visible to the person watching; with something like a message. Also, the idea of being able to say something as you bite the pill would be dope.

tumblr_mg6hyvMDms1qlu8two7_500

**Comment by LucyGrind** [comment]
oh apparently its usable while dead and i am a moron, do not merge.

File: code/modules/surgery/dental_implant.dm [hunk 1]
```diff
@@ -42,10 +42,10 @@
 	)
 	return ..()
 
-/datum/action/item_action/hands_free/activate_pill
+/datum/action/item_action/activate_pill
 	name = "Activate Pill"
-
-/datum/action/item_action/hands_free/activate_pill/Trigger(trigger_flags)
+	check_flags = NONE
+/datum/action/item_action/activate_pill/Trigger(trigger_flags)
```
Comment (MrMelbert) [inline code/modules/surgery/dental_implant.dm:48]: This also means it's usable while dead which is probably not idea

You can get around that with:
```dm
/datum/action/item_action/activate_pill/IsAvailable(feedback)
    if(owner.stat > SOFT_CRIT)
        return FALSE
    return ..() 
```

File: code/modules/surgery/dental_implant.dm [hunk 2]
```diff
@@ -42,10 +42,10 @@
 	)
 	return ..()
 
-/datum/action/item_action/hands_free/activate_pill
+/datum/action/item_action/activate_pill
 	name = "Activate Pill"
-
-/datum/action/item_action/hands_free/activate_pill/Trigger(trigger_flags)
+	check_flags = NONE
```
Comment (MrMelbert) [inline code/modules/surgery/dental_implant.dm:47]:
```suggestion
	check_flags = NONE

```

**Comment by Ryll-Ryll** [comment]
> > Not a fan of adding additional ways for people to stand back up and continue a fight that they've lost, especially when there's no visible message to people around them that you just bit a pill
> 
> I agree with this, make it visible to the person watching; with something like a message. Also, the idea of being able to say something as you bite the pill would be dope.
> 
> tumblr_mg6hyvMDms1qlu8two7_500 [ tumblr_mg6hyvMDms1qlu8two7_500 ](https://private-user-images.githubusercontent.com/53197594/338795224-1c25f769-1327-4e90-8cab-6d56f7fa033e.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxNTcxNDcsIm5iZiI6MTcxODE1Njg0NywicGF0aCI6Ii81MzE5NzU5NC8zMzg3OTUyMjQtMWMyNWY3NjktMTMyNy00ZTkwLThjYWItNmQ1NmY3ZmEwMzNlLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEyVDAxNDcyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQyMDhhMWYwNzg1MWU4MTVlNDk4NjAwMmQ1ZGE4NjA1OTlkN2IxOGU2MWM5OTY2MmYwYWY1YTM1ZDAxNGFiYWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NAwr3Xw-qzssiwm6vjyMGJl05X2L1N8qjO31BZu1PDk) [ ](https://private-user-images.githubusercontent.com/53197594/338795224-1c25f769-1327-4e90-8cab-6d56f7fa033e.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxNTcxNDcsIm5iZiI6MTcxODE1Njg0NywicGF0aCI6Ii81MzE5NzU5NC8zMzg3OTUyMjQtMWMyNWY3NjktMTMyNy00ZTkwLThjYWItNmQ1NmY3ZmEwMzNlLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEyVDAxNDcyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQyMDhhMWYwNzg1MWU4MTVlNDk4NjAwMmQ1ZGE4NjA1OTlkN2IxOGU2MWM5OTY2MmYwYWY1YTM1ZDAxNGFiYWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NAwr3Xw-qzssiwm6vjyMGJl05X2L1N8qjO31BZu1PDk)

These points would actually resolve all my issues with it, a visible message to people in a 5 tile radius when you trigger it of the person thrashing their jaw, and a 2-3 second delay before popping the pill would be fine by me

**Comment by MrMelbert** [comment]
You can put `Fixes #83839` in your PR body to close the issue when merged

File: code/modules/surgery/dental_implant.dm (hunk 2 above)
Comment (LucyGrind) [reply code/modules/surgery/dental_implant.dm:47]: what does this change?

**Comment by LucyGrind** [comment]
> oh apparently its usable while dead and i am a moron, do not merge.

fixed death pills

**Comment by LucyGrind** [comment]
> You can put `Fixes #83839` in your PR body to close the issue when merged

where do i add that under?

**Comment by ATH1909** [comment]
> These points would actually resolve all my issues with it, a visible message to people in a 5 tile radius when you trigger it of the person thrashing their jaw, and a 2-3 second delay before popping the pill would be fine by me

that delay is actually a pretty huge nerf to toothpills, especially suicide ones. sec can just... keep you from killing yourself by shuffling you around.

**Comment by LucyGrind** [comment]
> > These points would actually resolve all my issues with it, a visible message to people in a 5 tile radius when you trigger it of the person thrashing their jaw, and a 2-3 second delay before popping the pill would be fine by me
> 
> that delay is actually a pretty huge nerf to toothpills, especially suicide ones. sec can just... keep you from killing yourself by shuffling you around. unless the delay isn't interrupted by movement, which changes things.

I agree, ideally there would be no delay, a message isn't that big of a deal, but a good compromise between the supporters and detractors of critpills would be no interruption via movement imho, would have to see what more people want in the mean time.

**Comment by LucyGrind** [comment]
> as other people have said, should have a bubble alert (or normal) like "[name] clenches their teeth aggressively!" but instant use

I agree with this heavily

**Comment by SimplySmiley123** [comment]
> > > These points would actually resolve all my issues with it, a visible message to people in a 5 tile radius when you trigger it of the person thrashing their jaw, and a 2-3 second delay before popping the pill would be fine by me
> > 
> > 
> > that delay is actually a pretty huge nerf to toothpills, especially suicide ones. sec can just... keep you from killing yourself by shuffling you around. unless the delay isn't interrupted by movement, which changes things.
> 
> I agree, ideally there would be no delay, a message isn't that big of a deal, but a good compromise between the supporters and detractors of critpills would be no interruption via movement imho, would have to see what more people want in the mean time.

Wouldn't make sense for getting shuffled to make you unable to clench your teeth regardless to be honest.

**Comment by LucyGrind** [comment]
> > > Not a fan of adding additional ways for people to stand back up and continue a fight that they've lost, especially when there's no visible message to people around them that you just bit a pill
> > 
> > 
> > I agree with this, make it visible to the person watching; with something like a message. Also, the idea of being able to say something as you bite the pill would be dope.
> > tumblr_mg6hyvMDms1qlu8two7_500
> >       
> >         [
> >           
> >         
> >             tumblr_mg6hyvMDms1qlu8two7_500
> >           ](https://private-user-images.githubusercontent.com/53197594/338795224-1c25f769-1327-4e90-8cab-6d56f7fa033e.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxNTcxNDcsIm5iZiI6MTcxODE1Njg0NywicGF0aCI6Ii81MzE5NzU5NC8zMzg3OTUyMjQtMWMyNWY3NjktMTMyNy00ZTkwLThjYWItNmQ1NmY3ZmEwMzNlLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEyVDAxNDcyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQyMDhhMWYwNzg1MWU4MTVlNDk4NjAwMmQ1ZGE4NjA1OTlkN2IxOGU2MWM5OTY2MmYwYWY1YTM1ZDAxNGFiYWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NAwr3Xw-qzssiwm6vjyMGJl05X2L1N8qjO31BZu1PDk)
> >         
> >         
> >           
> >             
> >               
> >             
> >             
> >               
> >               
> >             
> >           
> >           [
> >             
> >               
> >             
> >           ](https://private-user-images.githubusercontent.com/53197594/338795224-1c25f769-1327-4e90-8cab-6d56f7fa033e.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxNTcxNDcsIm5iZiI6MTcxODE1Njg0NywicGF0aCI6Ii81MzE5NzU5NC8zMzg3OTUyMjQtMWMyNWY3NjktMTMyNy00ZTkwLThjYWItNmQ1NmY3ZmEwMzNlLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEyVDAxNDcyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQyMDhhMWYwNzg1MWU4MTVlNDk4NjAwMmQ1ZGE4NjA1OTlkN2IxOGU2MWM5OTY2MmYwYWY1YTM1ZDAxNGFiYWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NAwr3Xw-qzssiwm6vjyMGJl05X2L1N8qjO31BZu1PDk)
> >         
> >        [ tumblr_mg6hyvMDms1qlu8two7_500 ](https://private-user-images.githubusercontent.com/53197594/338795224-1c25f769-1327-4e90-8cab-6d56f7fa033e.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxNTcxNDcsIm5iZiI6MTcxODE1Njg0NywicGF0aCI6Ii81MzE5NzU5NC8zMzg3OTUyMjQtMWMyNWY3NjktMTMyNy00ZTkwLThjYWItNmQ1NmY3ZmEwMzNlLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEyVDAxNDcyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQyMDhhMWYwNzg1MWU4MTVlNDk4NjAwMmQ1ZGE4NjA1OTlkN2IxOGU2MWM5OTY2MmYwYWY1YTM1ZDAxNGFiYWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NAwr3Xw-qzssiwm6vjyMGJl05X2L1N8qjO31BZu1PDk)
> >       
> >         [
> >           
> >         
> >             tumblr_mg6hyvMDms1qlu8two7_500
> >           ](https://private-user-images.githubusercontent.com/53197594/338795224-1c25f769-1327-4e90-8cab-6d56f7fa033e.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxNTcxNDcsIm5iZiI6MTcxODE1Njg0NywicGF0aCI6Ii81MzE5NzU5NC8zMzg3OTUyMjQtMWMyNWY3NjktMTMyNy00ZTkwLThjYWItNmQ1NmY3ZmEwMzNlLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEyVDAxNDcyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQyMDhhMWYwNzg1MWU4MTVlNDk4NjAwMmQ1ZGE4NjA1OTlkN2IxOGU2MWM5OTY2MmYwYWY1YTM1ZDAxNGFiYWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NAwr3Xw-qzssiwm6vjyMGJl05X2L1N8qjO31BZu1PDk)
> >         
> >         
> >           
> >             
> >               
> >             
> >             
> >               
> >               
> >             
> >           
> >           [
> >             
> >               
> >             
> >           ](https://private-user-images.githubusercontent.com/53197594/338795224-1c25f769-1327-4e90-8cab-6d56f7fa033e.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxNTcxNDcsIm5iZiI6MTcxODE1Njg0NywicGF0aCI6Ii81MzE5NzU5NC8zMzg3OTUyMjQtMWMyNWY3NjktMTMyNy00ZTkwLThjYWItNmQ1NmY3ZmEwMzNlLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEyVDAxNDcyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQyMDhhMWYwNzg1MWU4MTVlNDk4NjAwMmQ1ZGE4NjA1OTlkN2IxOGU2MWM5OTY2MmYwYWY1YTM1ZDAxNGFiYWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NAwr3Xw-qzssiwm6vjyMGJl05X2L1N8qjO31BZu1PDk)
> >         
> >        [ ](https://private-user-images.githubusercontent.com/53197594/338795224-1c25f769-1327-4e90-8cab-6d56f7fa033e.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxNTcxNDcsIm5iZiI6MTcxODE1Njg0NywicGF0aCI6Ii81MzE5NzU5NC8zMzg3OTUyMjQtMWMyNWY3NjktMTMyNy00ZTkwLThjYWItNmQ1NmY3ZmEwMzNlLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEyVDAxNDcyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQyMDhhMWYwNzg1MWU4MTVlNDk4NjAwMmQ1ZGE4NjA1OTlkN2IxOGU2MWM5OTY2MmYwYWY1YTM1ZDAxNGFiYWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NAwr3Xw-qzssiwm6vjyMGJl05X2L1N8qjO31BZu1PDk)
> 
> These points would actually resolve all my issues with it, a visible message to people in a 5 tile radius when you trigger it of the person thrashing their jaw, and a 2-3 second delay before popping the pill would be fine by me

added everything you asked for, my only wish is you keep the 7.5 second hardcrit pill popping, although thats easy to revert if you insist

**Review by Watermelon914 — CHANGES_REQUESTED** [review]
This shouldn't be possible in hardcrit, but I'd be fine with it being done in softcrit.

**Comment by LucyGrind** [comment]
its finished :D merge whenever y'all want :3

**Pull request merged**